### PR TITLE
Bugfix/tlt 924 create button disabled adding intervention point

### PR DESCRIPTION
**Pull request by elliottyates**
fixes url validation inconsistency between element validation and custom validation regex; fixes disabled button issues on IP modal

### CODE REVIEW

File: ab_tool/static/ab_tool/js/experiments_dashboard.js [hunk 1]
```diff
@@ -1,17 +1,61 @@
...
+            $submit_button.siblings('.modal-cancel').addClass("disabled");
+        }
     });
 
     function is_valid_url(url) {
+        //if URL input element is used, this should match the pattern attribute or the custom validation
         return /^(http(s)?:\/\/)?(www\.)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i.test(url);
     }
 
+    function enable_submit_if_no_errors() {
```
Comment (roderickmorales) [inline ab_tool/static/ab_tool/js/experiments_dashboard.js]: For this to work you need to add .clean to ALL input fields. How you have it now, you are able to submit the form by just filling out the Name of the intervention. As far as I know we don't want to allow creation of intervention points with empty URLs. We should talk about making this work better.

Comment (elliottyates) [inline ab_tool/static/ab_tool/js/experiments_dashboard.js]: Roderick, perhaps I misunderstood the functionality then. We have a 'missing URLs' warning on the experiment dashboard, and the previous functionality allowed missing URLs, so I assumed that was part of the specifications. Happy to remove them of course if that's the way it should be. Thanks!

**Comment by roderickmorales** [comment]
we also need to account for the case when canceling the creation of a new IP. At the moment if you cancel or close the modal - all fields stay populated

**Comment by elliottyates** [comment]
@roderickmorales about the canceling -- this was also previous functionality, I think. If it should be cleared, let me know and I can fix it up, thanks.

**Comment by roderickmorales** [comment]
Same for editing - if you edit and IP and cancel/close the modal all values should be restored

**Comment by roderickmorales** [comment]
@elliottyates  so the canceling functionality was there? I though I had implemented it but wasn't sure since it was a little while ago. Either way we need that.

**Comment by elliottyates** [comment]
Ok so we've got the following functionality change to confirm:
- every update or create IP modal window open should present a fresh set of inputs
 
I think the missing URL thing was to allow people to fill in URLs later (e.g. TFs), no? That's why you can save IPs that way (but can't start an experiment until the URLs are filled in). Let's clarify that requirement today, we can talk about it.

**Comment by roderickmorales** [comment]
A fresh set of inputs when creating but when editing present the current data for those fields.

Missing URLs are allowed only if the user edits the experiment. Example: 
- You create an experiment with 2 tracks and create a bunch of IPs 
- After a while you edit your experiment by adding 1 more track, so you go from 2 tracks to 3 tracks
- At this point you will have missing URLs for all IPs created when your experiment only had 2 tracks

**Comment by elliottyates** [comment]
Got it thanks. I'll get back to work on this in the afternoon when we have our pow-wow.

**Pull request closed without merging**